pkg/apis/componentconfig/v1alpha1/defaults.go

```diff
@@ -120,41 +120,70 @@ func SetDefaults_KubeProxyConfiguration(obj *KubeProxyConfiguration) {
 }
 
 func SetDefaults_KubeSchedulerConfiguration(obj *KubeSchedulerConfiguration) {
-	if obj.Port == 0 {
-		obj.Port = ports.SchedulerPort
+	if len(obj.SchedulerName) == 0 {
+		obj.SchedulerName = api.DefaultSchedulerName
 	}
-	if obj.Address == "" {
-		obj.Address = "0.0.0.0"
+
+	if obj.HardPodAffinitySymmetricWeight == 0 {
+		obj.HardPodAffinitySymmetricWeight = api.DefaultHardPodAffinitySymmetricWeight
 	}
-	if obj.AlgorithmProvider == "" {
-		obj.AlgorithmProvider = "DefaultProvider"
+
+	if len(obj.AlgorithmSource.Type) == 0 {
+		obj.AlgorithmSource.Type = SchedulerAlgorithmSourceProvider
 	}
-	if obj.ContentType == "" {
-		obj.ContentType = "application/vnd.kubernetes.protobuf"
+
+	switch obj.AlgorithmSource.Type {
+	case SchedulerAlgorithmSourceProvider:
+		if obj.AlgorithmSource.Provider == nil || len(*obj.AlgorithmSource.Provider) == 0 {
+			val := SchedulerDefaultProviderName
+			obj.AlgorithmSource.Provider = &val
+		}
+	case SchedulerAlgorithmSourcePolicy:
+		// There is no default file or config map to apply.
```

Comment:
@bsalamat ^ Should we choose one?  The longstanding plan was to default to configmap.

Reply:
We do support a default namespace for the config map, but not a default name. A default name sounds good to me. Perhaps somebody else can think of a reason we shouldn't.

Reply:
When there is no policy file or policy ConfigMap, scheduler uses its builtin policy. We may change that one day, but that is beyond the scope of this PR.